server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -1200,6 +1264,34 @@ private static boolean isIndexMetadataPath(Path path) {
         return Files.isDirectory(path) && path.getFileName().toString().equals(MetadataStateFormat.STATE_DIR_NAME);
     }
 
+    /**
+     * Collect the path containing cache data in the indicated cache node path.
+     * The returned paths will point to the shard data folder.
+     */
+    static List<Path> collectCacheDataPath(NodePath cacheNodePath) throws IOException {
```

Comment:
Can this be private?

Reply:
This will be utilized by `org.opensearch.env.NodeRepurposeCommand` to calculate cache based paths for clean up, which is why package-private.